backend/LexBoxApi/GraphQL/UserMutations.cs

```diff
@@ -63,6 +74,50 @@ EmailService emailService
         return UpdateUser(loggedInContext, permissionService, input, dbContext, emailService);
     }
 
+    [Error<NotFoundException>]
+    [Error<DbError>]
+    [Error<UniqueValueException>]
+    [Error<RequiredException>]
+    [AdminRequired]
+    public async Task<User> CreateGuestUserByAdmin(
+        LoggedInContext loggedInContext,
+        CreateGuestUserByAdminInput input,
+        LexBoxDbContext dbContext
+    )
+    {
+        using var createGuestUserActivity = LexBoxActivitySource.Get().StartActivity("CreateGuestUser");
+
+        var hasExistingUser = input.Email is null && input.Username is null
```

Comment:
Supporting a username is going to mean extra work, but I agree that it makes sense, considering we're trying to extend/improve the experience of using the bulk tool.

I think there should probably just be one field in the UI for either the email or the username.
We're doing that in the Add-member modal and have validation that looks like this:
```
    usernameOrEmail: z.string().trim()
      .min(1, $t('project_page.add_user.empty_user_field'))
      .refine((value) => !value.includes('@') || isEmail(value), { message: $t('form.invalid_email') }),
```
(it's not complete enough for your case, because it doesn't have to validate usernames)

You could still split that field into 2 in the GQL input payload.

Anyway, I think there should only be 1 text field, however you end up doing this 🙂

Reply:
Admins can now create individual users with just a username in commit 6b19e6c2.

Reply:
Note that this doesn't validate the username, because we haven't decided on a rule for usernames. Should they be restricted to alphanumeric plus underscore only? Should uppercase be allowed? What about accents and/or non-Roman characters — what if someone wants a username in Thai or Korean?

Once we pick a rule, I'll implement that, but it's the end of the day and I'm not quite up to making that decision in my current state of mental fatigue. :-) Leaving this conversation unresolved for now, so that we can decide on what characters we'll allow in usernames.